### DISCUSSION TITLE
Dart language script support

### DISCUSSION THREAD

**Discussion by Ing-Brayan-Martinez** [godotengine/godot-proposals#9657] [Scripting]
Greetings, this is my first proposal, that's why I wanted to ask if support is possible for writing scripts in Dart language to develop in Godot.

Personally, I use this language every day thanks to Flutter. To put it into context, the Dart language is statically typed and has a fairly friendly syntax. It could be compared to Python. It allows interoperating with C++. Rust Objective C.

Dart is the language used for the Flutter framework which allows you to create cross-platform applications thanks to the Impeller graphics engine and its good compatibility for working with C++. This language is very versatile thanks to its syntax.

One day I asked myself the question if Dart language working with the Impeller graphics engine is excellent because it has been optimized for that purpose, why not use Dart language in other graphics engines so that they make intensive use of this language?

After thinking about it, I decided to create this proposal to find out if it is possible to do this implementation in Godot and consult with the community since I am not an expert on the internal architecture of Godot, so I am asking for help to understand how to make this possible.

Perhaps this proposal has little acceptance because the Dart language is an unknown technology for many developers and I think that through this proposal many developers will be encouraged to use it because they will be able to learn about the capabilities of this language.

**Comment by AThousandShips** [comment]
You can add it as an extension, but it's not likely to be added to the engine itself

**Comment by dsnopek** [comment]
There is a GDExtension that allows this:

https://github.com/fuzzybinary/godot_dart
